sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -649,7 +684,9 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
                 inputExprs.map(nodeToExpr),
                 unescapedScript,
                 output,
-                withWhere))
+                withWhere, inputFormat, outputFormat,
```

Comment:
I would like to put the `inputFormat`, `outputFormat` `inputSerDe` etc. into a single object, named `ScriptInputOutputSchema`, and the `SQLContext` and `HiveContext` may have different implementatio nof the `ScriptInputOutputSchema`. that's how the `Analyzer` of·SQLContext` or `HiveContext`  works (to resolve them, e.g. Schema extracting, SerDe class existence checking etc.).